shared/ui/components/form/inputs/OptionsInput.js

```diff
@@ -1,98 +1,83 @@
 import PropTypes from 'prop-types'
 import React from 'react'
 import styled from 'styled-components'
-import { Button, IconRemove, TextInput, theme, unselectable } from '@aragon/ui'
+import { Button, IconRemove, TextInput, useTheme, unselectable } from '@aragon/ui'
 
 const OptionsInput = ({
-  input,
-  name,
+  options,
   onChange,
-  placeholder = '',
-  validator,
-  values,
 }) => {
-  const validated = !input || validator(values, input.addr)
+  const theme = useTheme()
 
-  const addOption = () => {
+  const changeOption = (value, id) => {
     onChange({
-      target: validated
-        ? { name, value: [ ...values, input ] }
-        : { name: 'addressError', value: true }, // enable error msg if needed
+      target: { name: 'optionsChange', value, id },
     })
-    cleanInputBox()
-  }
-
-  const removeOption = option => {
-    // perform the change on the parent by using onChange prop without modifying value prop
-    option &&
-      onChange({ target: { name, value: values.filter(v => v !== option) } })
   }
 
-  const cleanInputBox = () => {
-    onChange({ target: { name: 'userInput', value: { addr: '' } } })
+  const addOption = () => {
```

Comment:
I think it would be great if we could focus the most recently added field, rather than requiring the user to click or tab backwards twice to focus it themselves. That might be out of scope for now, though. I'm having trouble thinking of an easy and error-free way of doing it.

Reply:
I did this by setting `autofocus` to true for every input except the first:
```
<StyledOption key={id}>
              <TextInput
                value={options[id]}
                onChange={v => changeOption(v.target.value, id)}
                wide
                autofocus={index !== 0}
              />
```
However, the only issue is when a user removes one of the options, the focus resets to the first option in the list. I'm not sure why that is the case. But it seems like such a small issue, I feel like its not worth putting the time in to solving it right now